File: codebase/code/ch10_30/ios/App/ViewModels/NotificationTokenViewModel.swift
import Foundation

class NotificationTokenViewModel {
    func register(_ token: String) async {
        let url = baseURL.appending(path: "notification_tokens")

        var req = URLRequest(url: url)
        req.httpMethod = "POST"
        req.setValue("application/json", forHTTPHeaderField: "Accept")
        req.setValue("application/json", forHTTPHeaderField: "Content-Type")
    }
}

File: codebase/code/ch10_61/android/app/src/main/java/com/masilotti/hikingjournal/viewmodels/NotificationTokenViewModel.kt
package com.masilotti.hikingjournal.viewmodels

import android.webkit.CookieManager
import com.masilotti.hikingjournal.activities.baseURL
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.net.HttpURLConnection
import java.net.URL

class NotificationTokenViewModel {
    suspend fun registerToken(token: String) = withContext(Dispatchers.IO) {
        val url = URL("$baseURL/notification_tokens")

        val connection = url.openConnection() as HttpURLConnection
        connection.requestMethod = "POST"
        connection.setRequestProperty("Content-Type", "application/json")

        CookieManager.getInstance().getCookie(baseURL)?.let {
            connection.setRequestProperty("Cookie", it)
        }
    }
}
